modules/apps/client-extension/client-extension-type-impl/src/main/java/com/liferay/client/extension/type/internal/configuration/CETConfigurationFactory.java

```diff
@@ -74,25 +108,112 @@ protected void modified(Map<String, Object> properties) throws Exception {
 				_cet = _cetManager.addCET(
 					cetConfiguration, companyId,
 					_getExternalReferenceCode(properties));
+
+				if (Objects.equals(
+						_cet.getType(),
+						ClientExtensionEntryConstants.TYPE_THEME_CSS)) {
+
+					_clientExtensionEntryRelLocalService.
+						deleteClientExtensionEntryRels(
+							companyId, _cet.getExternalReferenceCode());
+
+					_addThemeCSSClientExtensionEntryRel(companyId);
+				}
 			});
 	}
 
+	private void _addThemeCSSClientExtensionEntryRel(Long companyId)
+		throws PortalException {
+
+		ThemeCSSCET themeCSSCET = (ThemeCSSCET)_cet;
+
+		if (!Objects.equals(themeCSSCET.getScope(), "controlPanel")) {
+			return;
+		}
+
+		ClientExtensionEntryRel clientExtensionEntryRel =
+			_clientExtensionEntryRelLocalService.
+				fetchClientExtensionEntryRelByExternalReferenceCode(
+					themeCSSCET.getExternalReferenceCode(), companyId);
+
+		if (clientExtensionEntryRel != null) {
+			return;
+		}
+
+		Company company = _companyLocalService.getCompany(companyId);
+
+		User guestUser = company.getGuestUser();
+
+		Layout layout = _getControlPanelLayout(companyId);
+
+		_clientExtensionEntryRelLocalService.addClientExtensionEntryRel(
+			guestUser.getUserId(), layout.getGroupId(),
+			_portal.getClassNameId(Layout.class), layout.getPlid(),
+			_cet.getExternalReferenceCode(), _cet.getType(), StringPool.BLANK,
+			new ServiceContext());
+
+		int clientExtensionEntryRelsCount =
+			_clientExtensionEntryRelLocalService.
+				getClientExtensionEntryRelsCount(
+					_portal.getClassNameId(Layout.class), layout.getPlid(),
+					ClientExtensionEntryConstants.TYPE_THEME_CSS);
+
+		if (clientExtensionEntryRelsCount > 1) {
+			_log.error(
```

Comment:
@ethib137 

Drew's comment

> Also, do want also mention what the result is? That no theme CSS will be applied until one is removed?

What do you think?

Reply:
Yeah, that seems helpful.

Reply:
Nice! The one you picked is good.